### PR TITLE
stolostron/backlog#20552 add globalnet

### PR DESCRIPTION
**Pull request by cadawson**

### CODE REVIEW

File: add-ons/submariner/submariner.adoc [hunk 1]
```diff
@@ -104,3 +104,12 @@ After adding the utility to your path, the following table briefly describes the
 |===
 
 For more information about the `subctl` utility and its commands, see https://submariner.io/operations/deployment/subctl/[SUBCTL] in the Submariner documentation. 
+
+[#submariner-globalnet]
+== Globalnet
+
+Globalnet is a feature that can be applied to a cluster set that contains clusters that have the Submariner add-on enabled. Globalnet adds a global virtual network to the cluster set that resolves conflicts across clusters that are caused by overlapping CIDRs. Each cluster is given a subnet from the virtual global private network, configured as a new cluster parameter `GlobalCIDR`. The Globalnet controller uses the global CIDRs to support inter-cluster communication.
```
Comment (sridhargaddam) [inline add-ons/submariner/submariner.adoc]: Can you please see if you can phrase the sentence as follows...

Globalnet is a feature, part of Submariner add-on, which supports connectivity between clusters with overlapping CIDRs. Its a cluster set wide configuration and can be selected at the time the first managed cluster is added to the cluster set. When Globalnet is enabled, each managed cluster is allocated a global CIDR from the virtual Global Private Network. The global CIDR is used for supporting inter-cluster communication.

File: add-ons/submariner/submariner.adoc [hunk 2]
```diff
@@ -104,3 +104,12 @@ After adding the utility to your path, the following table briefly describes the
...
+
+You must have cluster administrator privileges to enable Globalnet. You can enable Globalnet for a cluster set by selecting the option to *Enable Globalnet* in the console when you add the Submariner add-on to clusters in the cluster set.
```
Comment (sridhargaddam) [inline add-ons/submariner/submariner.adoc]: Can you also include documentation on the Required access level as describe in point number 5 here - https://github.com/stolostron/backlog/issues/20552

Some additional info on the PR itself - https://github.com/stolostron/submariner-addon/pull/325

File: add-ons/submariner/submariner.adoc [hunk 3]
```diff
@@ -104,3 +104,14 @@ After adding the utility to your path, the following table briefly describes the
...
+
+If there is a chance that your clusters running Submariner might have overlapping CIDRs, consider enabling Globalnet. Globalnet is configured by creating a `submariner-broker` in the the `<ManagedClusterSet>-broker` namespace. The `ClusterAdmin` role has the required permissions to create this object in the broker namespace. The `ManagedClusterSetAdmin` role, which is sometimes created as a proxy administrator to act as an administrator for the cluster set, does not have the required permissions.). The `ClusterAdmin` creates the role and assigns it to the `ManagedClusterSetAdmin` user. 
+
+The `ClusterAdmin` role can enable Globalnet for a cluster set by selecting the option to *Enable Globalnet* in the console when enabling the Submariner add-on to clusters in the cluster set. After you enable it, you cannot disable it without removing Submariner.
```
Comment (sridhargaddam) [inline add-ons/submariner/submariner.adoc]:
```suggestion
If there is a chance that your clusters running Submariner might have overlapping CIDRs, consider enabling Globalnet. When using the ACM console, the `ClusterAdmin` can enable Globalnet for a cluster set by selecting the option *Enable Globalnet* when enabling the Submariner add-on to clusters in the cluster set. After you enable it, you cannot disable it without removing Submariner.

When using ACM APIs, Globalnet can be enabled by creating a `submariner-broker` object in the the `<ManagedClusterSet>-broker` namespace. The `ClusterAdmin` role has the required permissions to create this object in the broker namespace. The `ManagedClusterSetAdmin` role, which is sometimes created to act as a proxy administrator for the cluster set, does not have the required permissions. The `ClusterAdmin` is expected to associate the ClusterRole `access-to-brokers-submariner-crd` to the `ManagedClusterSetAdmin` user.

Sample `submariner-broker` object.
# Create the broker config specifying the Globalnet configuration
cat <<EOF | oc apply -f -
apiVersion: submariner.io/v1alpha1
kind: Broker
metadata:
  name: submariner-broker
  namespace: <enter-broker-namespace-here>
spec:
  globalnetEnabled: <true or false>
EOF
```

**Comment by sridhargaddam** [comment]
Thank you @cadawson, just some minor changes to the text and I felt providing an example might help the users.

**Comment by cadawson** [comment]
Thank you for the review @sridhargaddam . I made the updates, but made some minor changes to remain consistent with the other content in our docs. Let me know if anything needs to be updated. Thanks!

File: add-ons/submariner/submariner.adoc [hunk 4]
```diff
@@ -104,3 +104,41 @@ After adding the utility to your path, the following table briefly describes the
...
+
+[#submariner-globalnet]
+== Globalnet
+
+Globalnet is a feature included with the Submariner add-on which supports connectivity between clusters with overlapping CIDRs. Globalnet is a cluster set wide configuration, and can be selected when the first managed cluster is added to the cluster set. When Globalnet is enabled, each managed cluster is allocated a global CIDR from the virtual Global Private Network. The global CIDR is used for supporting inter-cluster communication.
+
+If there is a chance that your clusters running Submariner might have overlapping CIDRs, consider enabling Globalnet. When using the {product-title-short} console, the `ClusterAdmin` can enable Globalnet for a cluster set by selecting the option *Enable Globalnet* when enabling the Submariner add-on for clusters in the cluster set. After you enable Globalnet, you cannot disable it without removing Submariner.
+
+When using the {product-title-short} APIs, the `ClusterAdmin` can enable Globalnet by creating a `submariner-broker` object in the `<ManagedClusterSet>-broker` namespace. The `ClusterAdmin` role has the required permissions to create this object in the broker namespace.
```
Comment (sridhargaddam) [inline add-ons/submariner/submariner.adoc]:
```suggestion
When using the {product-title-short} APIs, the `ClusterAdmin` can enable Globalnet by creating a `submariner-broker` object in the `<ManagedClusterSet>-broker` namespace. 
```

This sentence is repeated.

**Review by sridhargaddam — COMMENTED** [review]
One final comment about a repetition, otherwise LGTM.

**Comment by cadawson** [comment]
Thanks @sridhargaddam . Good point. I made the change. Thanks!

**Pull request merged**